python/3d/auto_generated/qgspointcloudlayer3drenderer.sip.in

```diff
@@ -11,7 +11,7 @@
 
 
 
-class QgsPointCloudLayer3DRenderer : QgsAbstract3DRenderer
+class QgsPointCloudLayer3DRenderer : QObject, QgsAbstract3DRenderer
```

Comment:
Is there any particular reason why QgsPointCloudLayer3DRenderer is inheriting QObject ?

Reply:
Not really, that's a leftover! ;)

Reply:
Actually it needs to inherit QObject for the connect to work. I get build errors when trying with `QObject::connect()` and not inheriting QObject...

Reply:
That's why I'd move this up a level -- leave the guts of the "convert from 2d" in the 3d lib, but call it from layer level when the 2d renderer changes.

Reply:
I'm not sure how this would be done since the layer only has access to the `QgsAbstract3DRenderer`

Reply:
I would suggest:

1. Adding a subclass of QgsAbstract3DRenderer as QgsAbtract3DPointCloudRenderer
2. Add a 

    virtual bool setFrom2DRenderer( const QgsPointCloudRenderer* renderer ) = 0;

method to QgsAbtract3DPointCloudRenderer, and move your implementation of QgsPointCloudLayer3DRenderer::setSymbolFrom2DRenderer to the override of this pure virtual method. 

3. Move the "sync2DRendererTo3D" bool to QgsPointCloudLayer
4. Add logic in the QgsPointCloudLayer class to call setFrom2DRenderer and QgsMapLayer::setRenderer3D when the 2d renderer is changed.


That avoids the need for making the renderers a qobject, which is a no-no.

There's also existing logic in Qgs3DAppUtils::convert2dPointCloudRendererTo3d which does the 2d -> 3d conversion. This should be moved to a common place and reused here, so that we don't have the code duplication

Reply:
Ok, I'm working on implementing your suggestion.
I think the 3D renderer class should still be aware that it is synced from the 2d renderer and write/read that to the xml, so that we can still copy/paste a layer's style between layers while keeping the 3d renderer synced to the 2d on the pasted layer.